File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int res = 0;
        Set<Integer> set = new HashSet<>();

        for(int n : nums) {
            set.add(n);
        }

        for(int num : nums) {
            if(!set.contains(num - 1)) {
                int current = num;
                int length = 1;
                while(set.contains(current + 1)) {
                    current++;
                    length++;
                }
            res = Math.max(res, length);
            }
        }
        return res;

    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        curMax = 0
        curBuy = 100

        for num in prices:
            if num < curBuy:
                curBuy = num
            
            curMax = max(curMax, num - curBuy)
            
        return curMax